src/main.rs

```diff
@@ -1,4 +1,12 @@
+mod l1_fetcher;
+
 use clap::{arg, value_parser, Command};
+use ethers::types::U64;
+use eyre::Result;
+use l1_fetcher::L1Fetcher;
+use state_reconstruct::CommitBlockInfoV1;
+use std::sync::Arc;
```

Comment:
```suggestion
```
Is sync actually necessary here? Do we anticipate that we will share the vector of blocks anytime soon? Otherwise, I think just remove this for now.

Reply:
Individual Eth block contains multiple L2 blocks -> it's vector of blocks already, as I've understood.

Reply:
That might be true, but I still don't think we need to wrap our blocks in `Arc`. We don't yet share the value across any threads so we should be able to just use the raw value now, I think?

Reply:
> We don't yet share the value across any threads so we should be able to just use the raw value now, I think?
Actually, we do.

The L1 fetcher works in Tokio's async tasks and communicates results over the channel. My understanding is that Tokio runs these tasks in thread pool, unless you specifically configure it to disable the threading (which I think is undesired in our case). That then mandates that values involved are `Send`.

Is that not the case with Tokio & async?

Reply:
Well it depends on how the data is used within the actual tasks. Currently, as I see it, we have no actual concurrent use of the data (since it just gets sent to the sink and then processed from there). `Arc` is generally used for when we need to have *shared access* of a particular piece of data across threads/tasks, but we don't as we just transmit it across the task. Either way though, I don't hold a strong opinion of whether or not we keep as it, just pointing out an important distinction :wink: